xblock/fields.py

```diff
@@ -549,7 +558,7 @@ def __delete__(self, xblock):
         # Since we know that the field_data no longer contains the value, we can
         # avoid the possible database lookup that a future get() call would
         # entail by setting the cached value now to its default value.
-        self._set_cached_value(xblock, copy.deepcopy(self.default))
+        self._set_cached_value(xblock, self._get_default_value_to_cache(xblock))
```

Comment:
@cpennington IIRC, the test was failing due trying to deepcopy the `UNIQUE_ID` Sentinel. The `__get__` method above was using different logic to retrieve the default value, so I split that logic out and used it here as well. The test expected to default the cached value back to the same Sentinel default.